### PR TITLE
ADDED: support for docker + FIXED: issue can not read properties of u…

### PR DESCRIPTION
**Pull request by Mohammad-Bilal-Bhatti**
- Dockerize the application
- Added Support Docker Compose To run Application
- Fix can not read properties of undefined by reading length property when rending queues list
- Added localstack community docker image for ease
- Changed DEV_PORT from 10_007 to 3000
- Removed isProd environment checking - no use

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - DEBUG=${LOCALSTACK_DEBUG:-0}
+      - SERVICES=${LOCALSTACK_SERVICES}
+    networks:
+      - localnetwork
+    volumes:
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+
+networks:
+  localnetwork:
```
Comment (slaviqueue) [inline docker-compose.yml]: not sure if we have to spin localstack along with quick ui
I believe if someone uses this app, they already do have localstack installed

File: dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN mkdir -p /opt/app
+
+WORKDIR /opt/app
+
+COPY package.json package-lock.json .
+
+RUN npm install
+
+COPY . .
```
Comment (slaviqueue) [inline dockerfile:11]: this one will overwrite what we've just installed unless there's no `.dockerignore` file with `node_modules` added there

File: src/main.ts [hunk 3]
```diff
@@ -8,15 +8,14 @@ import { AppModule } from './app.module'
...
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  region: process.env.LOCALSTACK_REGION,
   logger: process.stdout,
 })
 
-const DEV_PORT = 10_007
-const isProd = process.env.NODE_ENV === 'prod'
-const serverPort = isProd ? 0 : DEV_PORT
+const appPort = parseInt(process.env.APP_PORT ?? '3000', 10);
```
Comment (slaviqueue) [inline src/main.ts]: I would do it this way
```suggestion
const appPort = Number(process.env.APP_PORT) ?? 3000;
```

File: src/sns/sns.service.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { PostMessageDTO } from './dto/post-message.dto'
 
 @Injectable()
 export class SnsService {
-  private readonly sns = new SNS({ endpoint: 'http://localhost:4566', region: 'us-east-1' })
+  private readonly sns = new SNS({ endpoint: `${process.env.LOCALSTACK_ENDPOINT}`, region: `${process.env.LOCALSTACK_REGION}` })
```
Comment (slaviqueue) [inline src/sns/sns.service.ts]: I would also move all these env variables to some config file

**Comment by slaviqueue** [comment]
Thanks a lot for you contribution! I've left a bunch of comments, you can update the PR and I will merge it.

File: docker-compose.yml (hunk 1 above)
Comment (Mohammad-Bilal-Bhatti) [reply docker-compose.yml]: yes you are right. I can make a separate docker-compose file to spin up localstack container. If some one want to spin localstack container they can do easily

File: dockerfile (hunk 2 above)
Comment (Mohammad-Bilal-Bhatti) [reply dockerfile:11]: we do have already .dockerignore file with node_modules init

**Comment by Mohammad-Bilal-Bhatti** [comment]
I have updated the PR and made required changes as suggested.

**Pull request merged**